sdk/storage/src/shared_access_signature/service_sas.rs

```diff
@@ -179,9 +182,63 @@ impl SasToken for BlobSharedAccessSignature {
             elements.push(format!("spr={protocol}"))
         }
 
+        if let Some(signed_directory_depth) = &self.signed_directory_depth {
+            elements.push(format!("sdd={signed_directory_depth}"))
+        }
+
         let sig = self.sign();
         elements.push(format!("sig={}", format_form(sig)));
 
         elements.join("&")
     }
 }
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use time::Duration;
+
+    const MOCK_SECRET_KEY: &str = "RZfi3m1W7eyQ5zD4ymSmGANVdJ2SDQmg4sE89SW104s=";
+    const MOCK_CANONICALIZED_RESOURCE: &str = "/blob/STORAGE_ACCOUNT_NAME/CONTAINER_NAME/";
+
+    #[test]
+    fn test_blob_scoped_sas_token() {
+        let permissions = BlobSasPermissions {
+            read: true,
+            ..Default::default()
+        };
+        let signed_token = BlobSharedAccessSignature::new(
+            String::from(MOCK_SECRET_KEY),
+            String::from(MOCK_CANONICALIZED_RESOURCE),
+            permissions,
+            OffsetDateTime::UNIX_EPOCH + Duration::days(7),
+            BlobSignedResource::Blob,
+        )
+        .token();
+        // BlobSignedResource::Blob
+        assert!(signed_token.contains("sr=b"));
+        // this assert will be a bad idea if we ever add a new field that ends with "sdd"
+        assert!(!signed_token.contains("sdd="));
```

Comment:
As this example uses a constant expire date of 1970-01-08, the signed token result should always be the same.  What are your thoughts for just having a const for the expected result?

Otherwise, to address your comment, this could also naively split on `&`, as the SasToken in the example should not result in a `&` except as a delim.

Reply:
Good idea!

I added both better splitting and an assert against a hardcoded string. This way you get more clear failures if things break but have a wider test too.

If people add more tests later or other things break the hardcoded tests they can remove one or the other